presto-tests/src/test/java/io/prestosql/execution/TestCoordinatorDynamicFiltering.java

```diff
@@ -0,0 +1,297 @@
+import io.prestosql.testing.TestingPageSinkProvider;
+import io.prestosql.testing.TestingTransactionHandle;
+import org.intellij.lang.annotations.Language;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Supplier;
+
+import static io.prestosql.SystemSessionProperties.JOIN_DISTRIBUTION_TYPE;
+import static io.prestosql.SystemSessionProperties.JOIN_REORDERING_STRATEGY;
+import static io.prestosql.spi.type.BigintType.BIGINT;
+import static io.prestosql.testing.TestingSession.testSessionBuilder;
+import static java.util.Collections.nCopies;
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.CompletableFuture.completedFuture;
+
+@Test(singleThreaded = true)
+public class TestCoordinatorDynamicFiltering
+        extends AbstractTestQueryFramework
+{
+    private final AtomicReference<TupleDomain<ColumnHandle>> expectedDynamicFilter = new AtomicReference<>(TupleDomain.all());
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        Session session = testSessionBuilder()
+                .setCatalog("test")
+                .setSchema("default")
+                .setSystemProperty(JOIN_REORDERING_STRATEGY, FeaturesConfig.JoinReorderingStrategy.NONE.name())
+                .setSystemProperty(JOIN_DISTRIBUTION_TYPE, FeaturesConfig.JoinDistributionType.PARTITIONED.name())
+                .build();
+        return DistributedQueryRunner.builder(session).build();
+    }
+
+    @BeforeClass
+    public void setUp()
+    {
+        getQueryRunner().installPlugin(new TestPlugin(expectedDynamicFilter));
+        getQueryRunner().installPlugin(new TpchPlugin());
+        getQueryRunner().createCatalog("test", "test", ImmutableMap.of());
+        getQueryRunner().createCatalog("tpch", "tpch", ImmutableMap.of());
+        computeActual("CREATE TABLE lineitem AS SELECT * FROM tpch.tiny.lineitem");
+    }
+
+    @AfterMethod(alwaysRun = true)
+    public void afterMethod()
+    {
+        expectedDynamicFilter.set(TupleDomain.all());
+    }
+
+    @Test
+    public void testWithEmptyBuildSide()
+    {
+        assertQueryDynamicFilters(
+                "SELECT * FROM lineitem JOIN tpch.tiny.supplier ON lineitem.suppkey = supplier.suppkey AND supplier.name = 'abc'",
+                TupleDomain.none());
+    }
+
+    @Test
+    public void testWithSelectiveBuildSide()
+    {
+        assertQueryDynamicFilters(
+                "SELECT * FROM lineitem JOIN tpch.tiny.supplier ON lineitem.suppkey = supplier.suppkey AND supplier.name = 'Supplier#000000001'",
+                TupleDomain.withColumnDomains(ImmutableMap.of(
+                        new TestingMetadata.TestingColumnHandle("suppkey", 2, BIGINT),
```

Comment:
make it `private static final` since it's used in multiple tests

Reply:
done